test/tabs/test-firefox-tabs.js

```diff
@@ -1261,6 +1261,40 @@ exports["test ready event after window.open"] = function (assert, done) {
   });
 }
 
+// related to bug #989288
+// https://bugzilla.mozilla.org/show_bug.cgi?id=989288
+exports["test tabs events after window.open"] = function(assert, done) {
+  // ensure popups open in a new windows and disable popup blocker
+  setPref(OPEN_IN_NEW_WINDOW_PREF, 2);
+  setPref(DISABLE_POPUP_PREF, false);
+
+  let counter = 0;
+  tabs.on('open', function onOpen(tab) {
+    assert.pass("tab open has occured");
+    counter++;
+    tabs.removeListener('open', onOpen);
+  });
```

Comment:
Using `once` you don't have to manually remove the listener at the end:
```js
tabs.once('open', (tab) => {
  assert.pass("tab open has occurred");
  counter++;
});
```
You could also transform this test in an async one, using a generator function:
```js
exports["test tabs events after window.open"] = function*(assert) {
  // prefs
  // open window to trigger observers
  tabs.activeTab.attach(…);

  let tab = yield wait(tabs, "open");
  assert.pass("tab open has occurred");

  yield wait(tabs, "ready");
  assert.pass("tab ready has occurred");
  
  tab.close();
  yield wait(tabs, "close");
  assert.pass("tab close has occurred");
}
```
Feel free to use what you feel more comfortable with.